yktrace/src/lib.rs

```diff
@@ -26,6 +29,16 @@ pub enum TracingKind {
     HardwareTracing
 }
 
+impl Default for TracingKind {
+    /// Returns the default tracing kind.
+    fn default() -> Self {
+        #[cfg(tracermode = "hw")]
+        return TracingKind::HardwareTracing;
+        #[cfg(tracermode = "sw")]
+        return TracingKind::SoftwareTracing;
```

Comment:
Maybe use `compile_error!` with a nice message when `-Ctracer` is not used.

Reply:
Using `compile_error!` seems to break doc tests and I don't seem to be able to fix it:

```
$ RUSTFLAGS="-C tracer=hw" cargo test
...
   Doc-tests yktrace
error: Please compile with `-C tracer=T`, where T is one of 'hw' or 'sw'
  --> /home/vext01/research/yorick/yk/yktrace/src/lib.rs:70:5
   |
70 |     compile_error!("Please compile with `-C tracer=T`, where T is one of 'hw' or 'sw'");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Despite:

```
#[cfg(all(doctest, not(any(tracermode = "hw", tracermode = "sw"))))]                           
compile_error!("Please compile with `-C tracer=T`, where T is one of 'hw' or 'sw'");           
```

Any ideas?

Reply:
Gah, I spotted it the moment I posted it.